### PR TITLE
Add Console FFI

### PR DESCRIPTION
**Pull request by hayesgm**
This patch adds the console FFI, allowing us to log messages from inside the EVM the same way that Forge/Foundry does.

The majority of this patch is simply handling Geno on some new corner cases (overloaded functions and void returns). These changes are a little hacky, but that's kinda the life of a code gen system.

### CODE REVIEW

File: Tests/GenoTests/Cool.swift [hunk 1]
```diff
@@ -7,76 +7,76 @@ public enum Cool {
...
+        } catch let EVM.QueryError.error(e, v) {
+            return .failure(rewrapError(e, value: v))
+        }
     }
 
     public static let vibeCheckFn = ABI.Function(
-            name: "vibeCheck",
-            inputs: [.uint256],
-            outputs: [.bool]
+        name: "vibeCheck",
```
Comment (coburncoburn) [inline Tests/GenoTests/Cool.swift:61]: is this new swift-format, or lacking one?

File: Sources/Eth/EVM.swift [hunk 2]
```diff
@@ -9,8 +9,26 @@ public enum EVM {
     static let sZero = BigInt(0)
     static let sOne = BigInt(1)
     static let wordZero = EthWord(fromBigInt: .zero)!
+    static let wordOne = EthWord(fromBigUInt: one)!
     static let gasAmount = EthWord(fromInt: 4_000_000)!
 
+    private static func consoleFFI(args: Hex) -> EVM.FFIResult {
+        let consoleFns: [ABI.Function] = IConsole.functions
+        let logFnDict = Dictionary(consoleFns.map { ($0.signatureHash, $0) }, uniquingKeysWith: { first, _ in first })
+        if args.count >= 4, let logCall = logFnDict[Hex(args.data.subdata(in: 0 ..< 4))] {
```
Comment (mykelp) [inline Sources/Eth/EVM.swift:18]: this is pretty sweet 💪

File: Sources/Geno/Geno.swift [hunk 3]
```diff
@@ -63,16 +63,44 @@ func createSourceFileSyntax(from contract: Contract, name: String, structsOnly:
...
+    var seenFunctions: [String: Int] = [:]
+    var res: [(Contract.ABI.Function, String)] = []
+
+    for functionPre in contractFunctions {
+        let function: Contract.ABI.Function
+        let count = seenFunctions[functionPre.name, default: 0]
+        if count > 0 {
+            function = Contract.ABI.Function(
+                type: functionPre.type,
+                name: "\(functionPre.name)\(count)",
```
Comment (coburncoburn) [inline Sources/Geno/Geno.swift:89]: maybe a separator would be good, since with say, logBytes2 & logBytes21 & logBytes22 it's not cleaer if we are logging 21 and 22 bytes or they are overloaded logBytes2 calls? like ` "\(functionPre.name)__\(count)"`?

Comment (hayesgm) [reply Sources/Geno/Geno.swift:89]: yeah, I thought about that, and yeah, you're probably right. Though best would be to get actual signature names on them.

**Review by coburncoburn — APPROVED** [review]
